packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -224,14 +231,20 @@ class Pigeon {
       final List<Method> functions = <Method>[];
       for (DeclarationMirror declaration in apiMirror.declarations.values) {
         if (declaration is MethodMirror && !declaration.isConstructor) {
+          final bool isAsynchronous =
```

Comment:
Can you switch this so that it is reading in the `async` keyword instead of using a using `@async` metadata?

Reply:
@gaaclarke by adding an `async` keyword to the method 
```dart
@HostApi
abstract class Api {
  Future<Foo> doit() async;
}
```
will look more like a broken code as it won't compile. Maybe we can try reading `Future<>` instead ?
```dart
@HostApi
abstract class Api {
  Future<Foo> doit();
}
```

Reply:
Why do you think that looks like broken code?  It looks correct to me.

Reply:
Because this doesn't compile.
```dart
@HostApi
abstract class Api {
  Future<Foo> doit() async;
}
```